crates/cli/src/config_file.rs

```diff
@@ -22,26 +20,106 @@ pub fn read_config_file(path: &str) -> Result<Option<ConfigFile>> {
+                _ => Err(anyhow!("cannot open config file")),
             }
         }
-        otherwise => otherwise?,
-    };
-    let mut contents = String::new();
-
-    let size_read = file
-        .read_to_string(&mut contents)
-        .context("error when reading the configration file")?;
-    if size_read == 0 {
-        return Err(anyhow!("the config file is empty"));
+        _ => Err(anyhow!("cannot open config file")),
+    }
+}
+
+// We first try to read static-analysis.datadog.yml
+// If it fails, we try to read static-analysis.datadog.yaml
+// If the file does not exist, we return a Ok(None).
+// If there is an error reading the file, we return a failure
+pub fn read_config_file(path: &str) -> Result<Option<ConfigFile>> {
+    if let Some(mut file) = get_config_file(path)? {
+        let mut contents = String::new();
+
+        let size_read = file
+            .read_to_string(&mut contents)
+            .context("error when reading the configration file")?;
+        if size_read == 0 {
+            return Err(anyhow!("the config file is empty"));
+        }
+        parse_config_file(&contents).map(Some)
+    } else {
+        Ok(None)
     }
+}
 
-    Ok(Some(parse_config_file(&contents)?))
+// Read the config file in base64
+pub fn read_config_file_in_base64(path: &str) -> Result<Option<String>> {
+    if let Some(mut file) = get_config_file(path)? {
+        let mut contents = String::new();
+
+        let size_read = file
+            .read_to_string(&mut contents)
+            .context("error when reading the configration file")?;
+        if size_read == 0 {
+            return Err(anyhow!("the config file is empty"));
+        }
+        Ok(Some(encode_base64_string(contents)))
+    } else {
+        Ok(None)
+    }
+}
+
+/// Get the final configuration for the analyzer
+/// First, try to get the configuration from the file
+/// - If the user is a Datadog user (e.g. with API keys), we fetch the remote configuration
+///   and merge it
+/// - If not, we just return the configuration
+pub fn get_config(path: &str, debug: bool) -> Result<Option<ConfigFile>> {
+    let config_file = read_config_file(path);
+    let repository_url_opt = get_repository_url(path);
+
+    config_file.map(|cf| {
+        if should_use_datadog_backend() && repository_url_opt.is_ok() {
+            let existing_config_file_base64 =
+                read_config_file_in_base64(path).expect("cannot get the config file in base64");
+            let remote_config = get_remote_configuration(
+                repository_url_opt.expect("repository URL should exist"),
+                existing_config_file_base64,
+                debug,
+            );
+
+            match remote_config {
+                Ok(rc) => {
+                    if debug {
+                        eprintln!("Remote config (base64): {:?}", rc);
+                    }
+
+                    let remote_config_base64 =
```

Comment:
I think you can swap the names of remote_config and remote_config_base64, right now remote_config is the base64 encoded version, and remote_config_base64 is normal text which is weird